### DISCUSSION TITLE
How to enable CacheLib as a secondary cache of RocksDB?

### DISCUSSION THREAD

**Discussion by DarthVaderSCY** [facebook/CacheLib#244] [Q&A] [answered]
Noticed that FB has open sourced the CacheLib implementation of Secondary Cache of RocksDB, but how can I integrate it into RocksDB? Could someone be kind enough to write a README with instructions on how to compile and enable it(such as running db_bench in RocksDB with a Secondary Cache pointing to CacheLib)?

**Comment by jaesoo-fb** [comment, marked as answer]
Hi @DarthVaderSCY 

AFAICT, @mrambacher was working on it in https://github.com/facebook/CacheLib/pull/184 It seems to be still WIP, but I think you can find the example and also README as well.

Also, please refer to https://rocksdb.org/blog/2021/05/27/rocksdb-secondary-cache.html and rocksdb issues like https://github.com/facebook/rocksdb/issues/8347 (CC @anand1976 )

**Reply by DarthVaderSCY** [reply]
Sorry to interrupt again, I'm very grateful for your answer, but what you mentioned above seem to be an incomplete WIP implementation and doesn't work for common user like me. And I notice that CacheLibWrapper is still being updated recently, so do you or someone involved in this mind sharing how FB inside manages to integrate this secondary cache into RocksDB? Because I'm interested in optimizing the data placement policy between Block Cache and secondary cache. Thanks very much!

**Comment by pbhandar2** [comment]
Closing this discussion as it has been inactive. If this is still relevant, then please create a new discussion post and we will get back to you as soon as we can. Thank you!
